Source/WebCore/page/SecurityOrigin.cpp

```diff
@@ -179,18 +160,30 @@ SecurityOrigin::SecurityOrigin(const SecurityOrigin* other)
 
 Ref<SecurityOrigin> SecurityOrigin::create(const URL& url)
 {
-    if (RefPtr<SecurityOrigin> cachedOrigin = getCachedOrigin(url))
-        return cachedOrigin.releaseNonNull();
+    if (url.protocolIsBlob())
+        return forBlobURL(url);
 
     if (SecurityOriginData::shouldTreatAsOpaqueOrigin(url))
         return adoptRef(*new SecurityOrigin);
 
-    if (shouldUseInnerURL(url))
-        return adoptRef(*new SecurityOrigin(extractInnerURL(url)));
-
     return adoptRef(*new SecurityOrigin(url));
 }
 
+Ref<SecurityOrigin> SecurityOrigin::forBlobURL(const URL& url)
+{
+    ASSERT(url.protocolIsBlob());
+
+    if (auto origin = getCachedOrigin(url))
+        return origin.releaseNonNull();
+
+    auto pathURL = URL({ }, url.path().toString());
+    // FIXME: blobinternal is web-exposed because of this
+    if (pathURL.isValid() && (pathURL.protocolIsInHTTPFamily() || pathURL.protocolIsFile() || pathURL.protocolIs("blobinternal"_s)))
```

Comment:
"`blobinternal`" might not be needed. That's the last thought I had for some `FileReader`-related failures I cannot figure out.

Reply:
I'm not certain, but the WK2 debug bot may give a hint (https://ews-build.s3-us-west-2.amazonaws.com/macOS-AppleSilicon-Ventura-Debug-WK2-Tests-EWS/c74bcce5-6818/results.html):
```
inspector-resource:///TestCombined.js:2719:30: CONSOLE JS ERROR Cannot load blob:inspector-resource:///324f619c-6e99-425a-a1a3-fe3d74081459.
```

In the stderr logs.

Reply:
Thank you, that helped a lot!